packages/shared/src/hooks/feed/useCardCover.tsx

```diff
@@ -0,0 +1,59 @@
+import React, { ReactNode, useMemo } from 'react';
+import { Post } from '../../graphql/posts';
+import { usePostShareLoop } from '../post/usePostShareLoop';
+import { CardCoverShare } from '../../components/cards/common/CardCoverShare';
+import { useBookmarkReminderEnrollment } from '../notifications';
+import { CardCoverContainer } from '../../components/cards/common/CardCoverContainer';
+import { PostReminderOptions } from '../../components/post/common/PostReminderOptions';
+import { ButtonSize, ButtonVariant } from '../../components/buttons/common';
+
+interface UseCardCover {
+  overlay: ReactNode;
+}
+
+interface UseCardCoverProps {
+  post: Post;
+  onShare: (post: Post) => void;
+}
+
+export const useCardCover = ({
+  post,
+  onShare,
+}: UseCardCoverProps): UseCardCover => {
+  const { shouldShowOverlay, onInteract } = usePostShareLoop(post);
+  const shouldShowReminder = useBookmarkReminderEnrollment(post);
+
+  const overlay = useMemo(() => {
+    if (shouldShowOverlay) {
```

Comment:
This one should take priority over the other one?

Reply:
Yes, as per [product's response](https://dailydotdev.slack.com/archives/C077EU278D7/p1719313817976609?thread_ts=1719313376.734949&cid=C077EU278D7).